config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/Entity.java

```diff
@@ -20,15 +20,17 @@
     private Element xml;
     private String name;
     private String originalName;
+    private boolean hasRouting;
     private String mappingAction;
     private Map<String, Object> mappingProperties = new HashMap<>();
 
     public Entity(String type, String name, String id, Element xml) {
-        this(type, name, null, id, xml, null);
+        this(type, name, name, id, xml, false);
     }
 
-    public Entity(String type, String originalName, String entityBundleName, String id, Element xml) {
+    public Entity(String type, String originalName, String entityBundleName, String id, Element xml, boolean hasRouting) {
```

Comment:
Accommodate all these unrelated/futuristic details under one arm.
Introduce properties like mappingProperties.
Use it to enable caller to provide all these extra details (entity-bundlename, hasRouting, etc) to the entity

Reply:
@graju256 done!